src/providers/ldap/sdap_fd_events.c

```diff
@@ -105,6 +107,19 @@ static int sdap_ldap_connect_callback_add(LDAP *ld, Sockbuf *sb,
         return EINVAL;
     }
 
+    ret = ldap_get_option(cb_data->sh->ldap, LDAP_OPT_NETWORK_TIMEOUT, &tv);
```

Comment:
If LDAP_OPT_NETWORK_TIMEOUT is set to `-1` which indicates that the timeout is infinite, `ldap_get_option()` will return `NULL` in `tv` (dont't ask me why) and `LDAP_OPT_SUCCESS`. Since `-1` is the default and  `sdap_ldap_connect_callback_add()` is also called during the setup of our normal LDAP connections, i.e. before we can set `LDAP_OPT_NETWORK_TIMEOUT`, this case must be handled as well. Currently I get crashes in `tv->tv_sec`.

Reply:
Done.